testplan/web_ui/testing/src/Report/BatchReportBeta/components/NavBreadcrumbWithNextRoute.jsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { Route } from 'react-router';
+import connect from 'react-redux/es/connect/connect';
+import _isString from 'lodash/isString';
+import _isObject from 'lodash/isObject';
+import _isArray from 'lodash/isArray';
+import uriComponentCodec from '../../../Common/uriComponentCodec';
+import { setHashComponentAlias } from '../state/uiActions';
+import { mkGetUIHashComponentToAlias } from '../state/uiSelectors';
+import NavBreadcrumb from './NavBreadcrumb';
+
+const connector = connect(
+  () => {
+    const getHashComponentToAlias = mkGetUIHashComponentToAlias();
+    return state => ({ hashComponentToAlias: getHashComponentToAlias(state) });
+  },
+  { boundSetHashComponentAlias: setHashComponentAlias },
+  (stateProps, dispatchProps, ownProps) => {
+    const { hashComponentToAlias } = stateProps;
+    const { boundSetHashComponentAlias } = dispatchProps;
+    const { entries, match } = ownProps;
+    const matchIsObj = _isObject(match);
+    const getTgtEntry = () => {
+      if(_isObject(entries)) {
+        if(_isArray(entries)) {
+          const encodedName = match.params.id;
+          // this incoming `encodedName` may be URL-encoded and so it
+          // won't match `entry.name` in the `entries` array, so we grab
+          // whatever `id` is actually an alias for, and use that to find
+          // our target `entry` object.
+          let decodedName = hashComponentToAlias[encodedName];
+          // on refresh on an aliased path, the `componentAliases` will
+          // be empty so we need to fill it with the aliased component
+          if(!decodedName) {
+            decodedName = uriComponentCodec.decode(encodedName);
+            boundSetHashComponentAlias(encodedName, decodedName);
+          }
+          return entries.find(e => decodedName === e.name);
+        } else {
+          return entries;
+        }
+      }
+      return null;
+    };
+    return { routePath: matchIsObj ? match.url : null, getTgtEntry };
+  },
+);
+
+const NavBreadcrumbWithNextRoute = connector(
+  ({ routePath, getTgtEntry }) => {
+    const tgtEntry = getTgtEntry();
+    const  tgtEntryIsObj = _isObject(tgtEntry);
+    const tgtEntryEntries = tgtEntryIsObj ? tgtEntry.entries : null;
+    return !(tgtEntryIsObj && _isString(routePath)) ? null : (
+      <>
+        <NavBreadcrumb entry={tgtEntry}/>
+        <Route path={`${routePath}/:id`}
+               render={({ match }) => (
+                 // this will render only if `${routePath}/:id` is matched
+                 <NavBreadcrumbWithNextRoute entries={tgtEntryEntries}
+                                             match={match}
+                 />
+               )}/>
+      </>
+    );
+  }
+);
+
```

Comment:
So this router and redux integration is mind boggling.
https://reactrouter.com/web/guides/deep-redux-integration